app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionOverlay.kt

```diff
@@ -49,8 +51,27 @@ class TrackingProtectionOverlay(
 
     @Suppress("MagicNumber", "InflateParams")
     private fun showTrackingProtectionOnboarding() {
+
         if (!getToolbar().hasWindowFocus()) return
 
+        val toolbarPosition = settings.toolbarPosition
+
+        when (toolbarPosition) {
+            ToolbarPosition.BOTTOM -> {
+                if(getToolbar().translationY > 0) {
+                    return
+                }
+            }
+            ToolbarPosition.TOP -> {
+                val appBarLayout = getToolbar().parent as AppBarLayout?
```

Comment:
One small thing but an important one:
`getToolbar().parent as AppBarLayout?` could still throw a `ClassCastException` if the parent is to be something different than `AppBarLayout?`.

Should we use safe cast here in the form
`getToolbar().parent as? AppBarLayout`
?

Reply:
Thanks for pointing it out. I'll push another commit.